### PR TITLE
Feature/idcom 1427  did compliance issuance

### PR DESCRIPTION
**Pull request by explicit-projects**
Changes:
* VC creation now accepts an issuer DID, verificationMethod and signing material (instead of the previous `CredentialSignerVerifier`)
* * A signer is either provided, or constructed from a provided keyPair/privateKey
* * The signer is provided to the CvcMerkleProof to sign the proof

[Additional Detail Here
](https://civicteam.atlassian.net/wiki/spaces/ID/pages/2396094508/Signing+Verifying+VCs+with+DIDs)
Additional Jest updates:
* added transpiling for certain packages (relating to digitalbazaar)
* converted `.babelrc` to `babel.config.js` as jest ignores `.babelrc`

Backwards compatibility:
* A "proxy" VC class that can handle accepting the old `CredentialSignerVerifier` for signing and verifying

Testing notes:
* The reasons for the fixtures for the different DIDs are commented near the top of `__test__/lib/util/did.js`
* Some TODOs to be removed once verify is implemented

### CODE REVIEW

File: __test__/services/configBranch.test.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 // do not put these tests on the same file as config.test.js or else jest will fail coverage
 // (describe.beforeEach scope not working)
 
+const originalPlatform = process.platform;
+
```
Comment (explicit-projects) [inline __test__/services/configBranch.test.js:5]: with the changes made in jest, it broke this functionality (using "this" in beforeEach is now undefined)

File: __test__/creds/VerifiableCredential.test.js [hunk 2]
```diff
@@ -1834,33 +1830,118 @@ describe('Transient Credential Tests', () => {
...
+      },
+    );
+
     expect(cred).toBeDefined();
-    expect(cred.proof.merkleRootSignature).toBeDefined();
-    expect(cred.verifyMerkletreeSignature(pubBase58)).toBeTruthy();
+    expect(cred.proof.merkleRootSignature.signature).toBeDefined();
+    expect(cred.proof.merkleRootSignature.verificationMethod).toBe(verificationMethod);
+
+    // TODO: re-activate this once verify is done
```
Comment (dankelleher) [inline __test__/creds/VerifiableCredential.test.js:1865]: is there a ticket for this? If so please add the number here so we know when to remove the todo

File: __test__/creds/VerifiableCredential.test.js [hunk 3]
```diff
@@ -1834,33 +1830,118 @@ describe('Transient Credential Tests', () => {
...
+        keypair,
+      },
+    );
+
+    expect(cred).toBeDefined();
+    expect(cred.proof.merkleRootSignature.signature).toBeDefined();
+    expect(cred.proof.merkleRootSignature.verificationMethod).toBe(verificationMethod);
+  });
+
+  // TODO: re-activate this once verify is done
```
Comment (dankelleher) [inline __test__/creds/VerifiableCredential.test.js:1919]: see above

File: __test__/services/configBranch.test.js (hunk 1 above)
Comment (dankelleher) [reply __test__/services/configBranch.test.js:5]: jest is weird. Maybe add a comment here for why we need to know the platform?

File: src/lib/signerVerifier.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+const signer = async (options) => {
+  if (!options.signer && !options.keypair && !options.privateKey) {
+    throw new Error('Either a signer, keypair or privateKey is required');
+  }
+
+  const { verificationMethod } = options;
+  let { signer: signerImpl } = options;
+
+  // Create a signer from keypair/key
+  if (!signerImpl) {
```
Comment (dankelleher) [inline src/lib/signerVerifier.js:45]: consider an early return instead of this if block
```suggestion
  if (signerImpl) return signerImpl
```

**Review by dankelleher — APPROVED** [review]

**Pull request merged**